### PR TITLE
feat: implement refresh contact method

### PR DESCRIPTION
**Pull request by gfanton**

### CODE REVIEW

File: api/protocoltypes.proto [hunk 1]
```diff
@@ -138,6 +138,9 @@ service ProtocolService {
 
   // PushSetServer registers a push server for the current device
   rpc PushSetServer(PushSetServer.Request) returns (PushSetServer.Reply);
+
+  // RefreshGroup try to force refresh the given group
```
Comment (aeddi) [inline api/protocoltypes.proto:142]: Should be named RefreshContactRequest

File: api/protocoltypes.proto [hunk 2]
```diff
@@ -138,6 +138,9 @@ service ProtocolService {
...
+  rpc RefreshRequest(RefreshRequest.Request) returns (RefreshRequest.Reply);
```
Comment (aeddi) [inline api/protocoltypes.proto:143]: Idem: should be named RefreshContactRequest

File: go/cmd/berty/mini/view_group_outgoing.go [hunk 3]
```diff
@@ -147,6 +147,11 @@ func commandList() []*command {
 			help:  "Shows IPFS debug information",
 			cmd:   debugIPFSCommand,
 		},
+		{
+			title: "refresh",
+			help:  "force refresh",
```
Comment (aeddi) [inline go/cmd/berty/mini/view_group_outgoing.go:152]: Can you name it `/contact refresh` please?

File: go/internal/tinder/driver_localdiscovery.go [hunk 4]
```diff
@@ -239,19 +250,19 @@ func (ld *LocalDiscovery) sendRecordsToProximityPeers(ctx context.Context, recor
...
-			if err := ld.sendRecordsTo(ctx, c.RemotePeer(), records); err != nil {
-				ld.logger.Error("unable to send records to newly connected peer",
-					zap.Error(err), zap.Stringer("peer", c.RemotePeer()))
-			}
+		if manet.IsPrivateAddr(c.RemoteMultiaddr()) || isProximityProtocol(c.RemoteMultiaddr()) {
+			wg.Add(1)
+			go func(c network.Conn) {
+				ld.logger.Debug("sending records to peer", zap.Stringer("peer", c.RemotePeer()))
+
+				if err := ld.sendRecordsTo(ctx, c.RemotePeer(), records); err != nil {
```
Comment (aeddi) [inline go/internal/tinder/driver_localdiscovery.go:258]: Can you put a `TODO` comment above this line about checking if we only send record of the contact request rendezvous and we filter out the groups rendezvous?

File: tool/berty-mini-local-helper/Makefile [hunk 5]
```diff
@@ -14,6 +14,11 @@ tail:
 	@touch $(TMPDIR)/berty-mini-$(ID).log
 	tail -f $(TMPDIR)/berty-mini-$(ID).log
 
+.PHONY: tail
```
Comment (aeddi) [inline tool/berty-mini-local-helper/Makefile:17]:
```suggestion
.PHONY: loon
```

**Review by aeddi — APPROVED** [review]
LGTM 👍

File: go/internal/tinder/driver_localdiscovery.go (hunk 4 above)
Comment (gfanton) [reply go/internal/tinder/driver_localdiscovery.go:258]: i've check and that the case: https://github.com/berty/berty/blob/11ede71bcb641a32edbfbf51de081c2213ab353b/go/internal/initutil/ipfs.go#L698

**Pull request merged**